bool avoid_activated
float64 tc
